### ISSUE TITLE
`IOFiberSpec` is still flaky in CI

### ISSUE THREAD

**Issue by djspiewak** [typelevel/cats-effect#3463]
```
[error]   x toString a completed fiber
[2135](https://github.com/typelevel/cats-effect/actions/runs/4284923854/jobs/7462682986#step:23:2136)
[error]    'cats.effect.IOFiber@4ad52419 RUNNING' doesn't match 'cats.effect.IOFiber@[0-9a-f][0-9a-f]+ COMPLETED' (IOFiberSpec.scala:68)
[2136](https://github.com/typelevel/cats-effect/actions/runs/4284923854/jobs/7462682986#step:23:2137)
```

We should consider disabling it in CI or something because we can't have jobs randomly failing like this.

**Comment by durban** [comment]
It seems this is the test: https://github.com/typelevel/cats-effect/blob/series/3.4.x/tests/shared/src/test/scala/cats/effect/IOFiberSpec.scala#L59-L70

The intention of this test is unclear to me. It seems to intentionally create a race condition? Why not `cancel` (or even `join`) the fiber, and _then_ call `toString`?

Also, `isDone` seems to have a racy read of `resumeTag`. Is this for performance? I think reading `outcome` would be essentially the same, but not-racy (as `outcome` is `volatile`).

**Comment by armanbilge** [comment]
Huh, `join`ing it seems fine. Not sure why that test was written like that ...

Your `isDone` suggestion makes sense to me as well.

**Comment by durban** [comment]
Instead of opening a new ticket: `"toString a suspended fiber"` hangs on `series/3.x`, as `async_` became uncancelable. But somehow the test doesn't fail. I don't understand specs2, but it marks the test with a purple `o` (instead of a green `+`).

**Comment by armanbilge** [comment]
Lovely 😂  well, let's fix that too :) just needs to be re-written with `async`.

**Comment by durban** [comment]
Fixed by #3467.
